angr/sim_state.py

```diff
@@ -436,6 +436,20 @@ def step(self, **kwargs):
         """
         return self.project.factory.successors(self, **kwargs)
 
+    def detect_loops(self, n=None):
+        mc1 = self.callstack.top.block_counter.most_common()
+        mc2 = collections.Counter([l for l in self.history.descriptions]).most_common()
```

Comment:
Directly accessing `self.history.descriptions` makes me feel sad. `history.descriptions` is a collection of string representations of `SimSuccessors`. They are meant to be read and used by humans. You should consider using `history.bbl_addrs` instead. At least, you avoid performing many unnecessary string comparisons.